airgun/session.py

```diff
@@ -161,33 +162,99 @@ def __init__(self, session_name=None, user=None, password=None, session_cookie=N
             passed - default one from settings will be used.
         :param requests.sessions.Session optional session_cookie: session object to be used
             to bypass login
+        :param str optional url: URL path to open when starting session (without protocol
+            and hostname)
         """
         self.name = session_name or gen_string('alphanumeric')
         self._user = user or settings.satellite.username
         self._password = password or settings.satellite.password
         self._session_cookie = session_cookie
         self._factory = None
+        self._url = None
```

Comment:
Shouldn't it take the `url` passed as an argument instead of `None`?
```suggestion
        self._url = url
```

Reply:
Right, of course. Thanks for catching this.
Changed in 140334a39bd280d575a6bf2d212766ecc0d54d96 . Yes, I do expect these to be squashed before merge :)